src/restconfig/src/main/java/org/geoserver/rest/CatalogReloadController.java

```diff
@@ -33,5 +35,8 @@ public void reload() throws Exception {
             method = {RequestMethod.POST, RequestMethod.PUT})
     public void reset() {
         geoServer.reset();
+        GeoServerExtensions.bean(GeoServerSecurityManager.class)
+                .getAuthenticationCache()
+                .removeAll();
```

Comment:
This looks much better indeed, but... it's still backwards.
See how the controller is ust calling the reset method, without any explicit, special action targeting a particular cache? E.g., it's not calling ResourcePool to reset its caches, it's the pool that' listening to the reset events and cleaning itself. I believe you can have the security manager implement GeoServerLifecycleHandler and react on the reset "onReset" event instead. Or have you already tried it, and it did not work?

Reply:
Implemented, successfully too, as you suggested (I think), but I am getting test errors in AuthenticationCacheFilterTest.java: 

> Caused by: org.geoserver.platform.GeoServerExtensions$MultipleBeansException: Multiple beans of type [org.geoserver.security](http://org.geoserver.security/).auth.AuthenticationCache


Clearly I still do not know how to instantiate the bean correctly.